src/hotspot/share/runtime/handshake.cpp

```diff
@@ -251,8 +251,13 @@ class VM_HandshakeAllThreads: public VM_Operation {
       thr->handshake_state()->add_operation(_op);
       number_of_threads_issued++;
     }
+
+    // Separate the arming of the poll in add_operation() above from
+    // the read of JavaThread state in the try_process() call below.
     if (UseSystemMemoryBarrier) {
       SystemMemoryBarrier::emit();
+    } else {
+      OrderAccess::fence(); // storestore|storeload, global state -> local state
```

Comment:
The comment is somewhat confusing as a fence is logically all four of the storeload etc barriers not just the pair listed. I suggest just removing the comment. And yes I realize this was copied from the safepoint code (but it also has a comment on the emit that is not here, so it is okay if they differ).

Reply:
+1 on removing this comment. Also seems global state refers to _state for the safepoint code which doesn't apply here.

Reply:
I copied this from the safepoint case.  Will remove.